### PR TITLE
Document how to use the OpenShift certificate manager

### PR DESCRIPTION
**Pull request by cescoffier**
OpenShift comes with its own version of the cert manager, based on annotations. See https://docs.openshift.com/container-platform/4.16/security/certificates/service-serving-certificate.html.

This PR extends the documentation to cover this OpenShift feature.

### CODE REVIEW

File: docs/src/main/asciidoc/tls-registry-reference.adoc [hunk 1]
```diff
@@ -520,12 +521,13 @@ When the application starts, the TLS registry performs some checks to ensure the
...
 If any of these checks fail, the application will fail to start.
 
+[#reloading-certificates]
 == Reloading Certificates
 
 The `TlsConfiguration` obtained from the `TLSConfigurationRegistry` includes a mechanism for reloading certificates.
 The `reload` method refreshes the key stores and trust stores, typically by reloading them from the file system.
 
-NOTE: The reload operation is not automatic and must be triggered manually. Additionally, the `TlsConfiguration` implementation must support reloading (which is the case for the configured certificate).
+NOTE: The reload operation is not automatic and must be triggered manually.Additionally, the `TlsConfiguration` implementation must support reloading (which is the case for the configured certificate).
```
Comment (gsmet) [inline docs/src/main/asciidoc/tls-registry-reference.adoc:530]: You need to revert this change.

File: docs/src/main/asciidoc/tls-registry-reference.adoc [hunk 2]
```diff
@@ -693,6 +695,227 @@ To handle the renewal, you can use the periodic reloading mechanism:
 %prod.quarkus.http.insecure-requests=disabled
 ----
 
+== Utilizing OpenShift Serving Certificates
```
Comment (gsmet) [inline docs/src/main/asciidoc/tls-registry-reference.adoc:698]: I think the rule is now to avoid capitalizing titles but I see the doc is all like that so probably good for this patch.

**Review by gsmet — CHANGES_REQUESTED** [review]
Nice content!

Not sure why you dropped this space but it needs to be added back.

For the titles, probably something to do in another PR someday, I just wanted to draw your attention on it for the future.

Please dismiss when space if fixed!

File: docs/src/main/asciidoc/tls-registry-reference.adoc (hunk 1 above)
Comment (cescoffier) [reply docs/src/main/asciidoc/tls-registry-reference.adoc:530]: That's me hitting a key to shutdown my screen saver...

File: docs/src/main/asciidoc/tls-registry-reference.adoc (hunk 2 above)
Comment (cescoffier) [reply docs/src/main/asciidoc/tls-registry-reference.adoc]: Fixed everywhere.
